go.mod

```diff
@@ -1,8 +1,6 @@
 module github.com/spegel-org/spegel
 
-go 1.24.0
-
-toolchain go1.25.1
```

Comment:
We don't want to change this. The toolchain version tells Github actions and also your local go tooling which Go version to use when building Go, while the Go version indicates the minimum version required when importing Spegel as a library.  AS we are not using any Go 1.25 specific language features we want to keep the minimum version where it is as k3s import Spegel as a library.

Reply:
This is done automatically by go mod tidy when changing the module version to 1.25.1, and the build won't succeed otherwise. The only way to prevent it is to have different versions for the module and the tool chain.

Reply:
Happy to keep the module version as Go 1.24 if you think it's the right thing to do, though bear in mind there may be language specific changes between major versions which the toolchain directive won't apply. The most obvious example I can think of is with for loop scopes. 

https://go.dev/blog/loopvar-preview

I guess the GOMAXPROCS changes aren't a language change so doesn't apply here though.